### ISSUE TITLE
False positive: const_static_lifetime

### ISSUE THREAD

**Issue by myrrlyn** [rust-lang-nursery/rust-clippy#2438]
```sh
$ rustc --version
rustc 1.25.0-nightly (0c6091fbd 2018-02-04)
$ cargo clippy --version
0.0.186
```

Running `cargo clippy` over my project with `#![cfg_attr(feature = "cargo-clippy", deny(clippy))]` set results in the following errors:

```text
error: Constants have by default a `'static` lifetime

    const KEYWORD: &'static str = "whatever";
                   -^^^^^^^----- help: consider removing `'static`: `&str`
```

Doing so promptly results in

```text
error[E0106]: missing lifetime specifier

    const KEYWORD: &str = "whatever";
                   ^ expected lifetime parameter
```

I presume that following Clippy's advice and running into a compiler-enforced hard error is a bug in Clippy?

In this particular case, the const in question is a trait associated constant:

```rust
trait Keyword {
    const KEYWORD: &'static str;
}
```

and the positions it is reporting for change are implementations of it:

```rust
impl Keyword for Whatever {
    const KEYWORD: &'static str = "whatever";
}
```

Removing any `'static` marker, whether in the trait declaration or in implementations of it, causes E0106 to occur.

**Comment by oli-obk** [comment]
We just need to skip impl and trait items
